src/Humanizer/Inflections/Vocabulary.cs

```diff
@@ -0,0 +1,172 @@
+    /// At this time, multiple vocabularies and removing existing rules are not supported.
+    /// </summary>
+    public class Vocabulary
+    {
+        private readonly List<Rule> _plurals = new List<Rule>();
+        private readonly List<Rule> _singulars = new List<Rule>();
+        private readonly List<string> _uncountables = new List<string>();
+
+        /// <summary>
+        /// Adds a word to the vocabulary which cannot easily be pluralized/singularized by RegEx, e.g. "person" and "people".
+        /// </summary>
+        /// <param name="singular">The singular form of the irregular word, e.g. "person".</param>
+        /// <param name="plural">The plural form of the irregular word, e.g. "people".</param>
+        public void AddIrregular(string singular, string plural)
+        {
+            AddPlural("(" + singular[0] + ")" + singular.Substring(1) + "$", "$1" + plural.Substring(1));
+            AddSingular("(" + plural[0] + ")" + plural.Substring(1) + "$", "$1" + singular.Substring(1));
+        }
+
+        /// <summary>
+        /// Adds an uncountable word to the vocabulary, e.g. "fish".  Will be ignored when plurality is changed.
+        /// </summary>
+        /// <param name="word">Word to be added to the list of uncountables.</param>
+        public void AddUncountable(string word)
+        {
+            _uncountables.Add(word.ToLower());
+        }
+
+        /// <summary>
+        /// Adds a rule to the vocabulary that does not follow trivial rules for pluralization, e.g. "bus" -> "buses"
+        /// </summary>
+        /// <param name="rule">RegEx to be matched, case insensitive, e.g. "(bus)es$"</param>
+        /// <param name="replacement">RegEx replacement  e.g. "$1"</param>
+        public void AddPlural(string rule, string replacement)
+        {
+            _plurals.Add(new Rule(rule, replacement));
+        }
+
+        /// <summary>
+        /// Adds a word to the vocabulary that does not follow trivial rules for singularization, e.g. "vertices/indices -> "vertex/index"
+        /// </summary>
+        /// <param name="rule">RegEx to be matched, case insensitive, e.g. ""(vert|ind)ices$""</param>
+        /// <param name="replacement">RegEx replacement  e.g. "$1ex"</param>
+        public void AddSingular(string rule, string replacement)
+        {
+            _singulars.Add(new Rule(rule, replacement));
+        }
+
+        /// <summary>
+        /// Pluralizes the provided input considering irregular words
+        /// </summary>
+        /// <param name="word">Word to be pluralized</param>
+        /// <param name="plurality">Normally you call Pluralize on singular words; but if you're unsure call it with Plurality.CouldBeEither</param>
+        /// <returns></returns>
+        [Obsolete("Use string.Pluralize(bool) instead. This method will be removed in next major release.")]
+        public string Pluralize(string word, Plurality plurality)
```

Comment:
These `Pluralize` and `Singularize` methods were initially created in Humanizer but then we introduced the other overloads and flagged these as obsolete. There may be users out there who still use these methods. I will put them back.